doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,257 @@
+.. _functional_in_depth:
+
+#####################
+Functional
+#####################
+
+A *functional* is an operator ``f`` that maps from some vector space ``X`` to the field of scalars ``F`` associated with the vector space:
+
+.. math::
+   
+   f : X \to F.
+
+In the ODL solver package, functionals are implemented in the `Functional` class, a subclass to `Operator`.
+
+From a mathematical presepective, the above is a valide definition of a functional.
+However, since the purpose of these functionals are primarely to be used for solving optimization problems, the following assumptions are made:
+
+ * the vector space ``X`` is a Hilbert space.
+ * the field of scalars ``F`` are the real numbers.
+
+The first assumption is made in order to simplify the concept of *convex conjugate functional*, see ``conjugate_functional`` under :ref:`implementation` for more details, or the Wikipedia articles on `convex conjugates`_ and `Legendre transformations`_.
+
+The second assumption is made in order to guarantee that the optimization problem, in some way, makes sense.
+For example, if the field is not (totally) ordered (like for example the complex numbers) one cannot define optimization over it in a meaningful way.
+Or if the field does not have the least-upper-bound property, then attainment of an optimal solution cannot be guaranteed.
+See, for example, the Wikipedia articles on `field`_, `ordered field`_ and `least-upper-bound property`_.
+
+Note that it is possible to use the class without fulfilling these assumtptions, however in this case some of the mathematical results might not hold.
+
+.. _convex conjugate: https://en.wikipedia.org/wiki/Convex_conjugate
+.. _Legendre transformation: https://en.wikipedia.org/wiki/Legendre_transformation
+
+.. _field: https://en.wikipedia.org/wiki/Field_(mathematics)
+.. _ordered field: https://en.wikipedia.org/wiki/Ordered_field
+.. _least-upper-bound property: https://en.wikipedia.org/wiki/Least-upper-bound_property
+
+.. _implementation:
+
+Implementation of functionals
+=============================
+
+To define your own functional, you start by writing::
+
+    class MyFunctional(odl.solvers.Functional):
+        ...
+
+Since `Functional` is a subclass of `Operator`, it has the *abstract method* ``domain`` which need to be explicitly overridden by any subclass.
+
+``domain``: `Set`
+    The domain of this functional, i.e., the set of elements to which this functional can be applied.
+
+Moreover, there are several optional parameters associated with a functional. These are
+
+``linear``: `bool`, optional
+    If `True`, the functional is considered as linear.
+    Default: ``False``
```

Comment:
I'd go on with the example for this property and the `grad_lipschitz` one, instead of this listing. 

Not yet sure where to have the list of parameters, perhaps in the class docstring of `Functional`, just without the surrounding explanations. I'll thing about it.